### DISCUSSION TITLE
how to compose the train_dataset with two different pipelines

### DISCUSSION THREAD

**Discussion by hadesfgh** [open-mmlab/mmediting#1346] [How-to] [answered]
[English] In mmediting, data.train in config (or data.val、data.test) has corresponding pipeline: train_pipeline. Can the data.train consist of two different datasets, which have corresponding pipeline .
[Chinese] 在配置文件中，一个数据集只能对应一个pipeline，即该数据集经过了对应的pipeline进行处理，不知道能否让训练集的一部分数据经过一个pipeline，另一部分数据经过一个另pipeline,亦或者让两个数据集经过对应的pipeline后最终合并成一个训练集。

**Comment by zengyh1900** [comment]
Hi, 

Do you mean using two pipelines for a single dataloader?
May I know in which situation you need it？

**Reply by hadesfgh** [reply]
Thank you for your reply, situation i need as follows:
Can you make one part of the data of the training set pass through one pipeline, and the other part of the data pass through another pipeline, or can you make two datasets finally merge into one training set after passing through the corresponding pipeline.

**Comment by LeoXing1996** [comment, marked as answer]
@hadesfgh , If the transform is randomly applied to each samples, you can use [`RandomApply`](https://github.com/open-mmlab/mmcv/blob/2.x/docs/zh_cn/understand_mmcv/data_transform.md#%E9%9A%8F%E6%9C%BA%E9%80%89%E6%8B%A9randomchoice%E5%92%8C%E9%9A%8F%E6%9C%BA%E6%89%A7%E8%A1%8Crandomapply) transform.

If you want to apply specific transforms to specific samples, you can decide whether you want to perform a data transformation operation based on the file name (or other metainfo). A demo implementation is provided as follow:
```python
class SampleSpecTransform(BaseTransform):
    def __init__(self, filter_key, *args, **kwargs):

        self.filter_key = filter_key  # a filter function to judge the sample
        ...

    def transform(self, results):
        # get some metainfo in some way
        img_path = results['img_path']

        # judge whether apply the transformation
        if self.filter_key in img_path:
            results = self._do_transform(results)

        return results
```
